tests/training/metrics_test.py

```diff
@@ -211,3 +213,89 @@ def test_f1_measure_works_for_sequences(self):
         numpy.testing.assert_almost_equal(precision, 1.0)
         numpy.testing.assert_almost_equal(recall, 0.5)
         numpy.testing.assert_almost_equal(f1, 0.66666666666)
+
+
+class ConllSpanBasedF1Test(AllenNlpTestCase):
+
+    def setUp(self):
+        super(ConllSpanBasedF1Test, self).setUp()
+        vocab = Vocabulary()
+        vocab.add_token_to_namespace("O", "tags")
+        vocab.add_token_to_namespace("B-ARG1", "tags")
+        vocab.add_token_to_namespace("I-ARG1", "tags")
+        vocab.add_token_to_namespace("B-ARG2", "tags")
+        vocab.add_token_to_namespace("I-ARG2", "tags")
+        self.vocab = vocab
+
+    def test_conll_span_based_f1_extracts_correct_spans(self):
+        o_index = self.vocab.get_token_index("O", "tags")
+        tag_vocab = self.vocab.get_index_to_token_vocabulary("tags")
+        metric = ConllSpanBasedF1Measure(tag_vocab, ignore_classes=[o_index])
+
+        tag_sequence = ["O", "B-ARG1", "I-ARG1", "O", "B-ARG2", "I-ARG2", "B-ARG1", "B-ARG2"]
+        indices = [self.vocab.get_token_index(x, "tags") for x in tag_sequence]
+        spans = metric._extract_spans(indices)
+        assert spans == {((1, 2), "ARG1"), ((4, 5), "ARG2"), ((6, 6), "ARG1"), ((7, 7), "ARG2")}
+
+        # Check that invalid BIO sequences are handled as non-spans.
+        tag_sequence = ["O", "B-ARG1", "I-ARG1", "O", "I-ARG1", "B-ARG2", "I-ARG2", "B-ARG1", "I-ARG2"]
```

Comment:
Hmm, seems like the model should be penalized in precision if it gives invalid BIO sequences.  I'd probably go the other way and count invalid tags like this as predictions.  Otherwise, if I have a predicted tag sequence that differs from a gold tag sequence only by replacing an "O" somewhere with an "I-ARG1", I'll get a perfect F1 score, when I really shouldn't.